learn/sherif/main_test/src/main/java/frc/robot/commands/Shoot.java

```diff
@@ -0,0 +1,31 @@
+package frc.robot.commands;
+
+import frc.robot.helpers.TestableCommand;
+import frc.robot.io.Slider;
+import frc.robot.subsystems.Factory;
+import frc.robot.subsystems.ShooterController;
+
+import javax.annotation.Nullable;
+
+public class Shoot extends TestableCommand {
+    private final ShooterController shooterController;
+    @Nullable private Slider speedSlider = null;
+
+    public Shoot(ShooterController shooterController) {
+        this.shooterController = shooterController;
+    }
+
+    public void useSlider(Factory f) {
+        this.speedSlider = f.getSlider("Shooter speed: ", 1.0, 0.0, 1.0);
+    }
```

Comment:
Shouldn't this be a part of the constructor?

Reply:
I put it in a separate function so it could be optional, since commands shouldn't normally require a factory